src/main/java/org/gridsuite/modification/server/service/LineTypesCatalogService.java

```diff
@@ -43,14 +54,20 @@ public void deleteLineTypesCatalog() {
         lineTypesCatalogRepository.deleteAll();
     }
 
-    public void resetLineTypes(List<LineTypeInfos> lineTypes) {
-        deleteLineTypesCatalog();
-        // remove duplicates in file
-        Set<LineTypeInfos> lineTypesSet = lineTypes.stream().collect(Collectors.toSet());
+    public void resetLineTypes(MultipartFile file) {
+        try (GZIPInputStream gzipInputStream = new GZIPInputStream(file.getInputStream())) {
+            List<LineTypeInfos> lineTypes = mapper.readValue(gzipInputStream, new TypeReference<>() {
+            });
+            deleteLineTypesCatalog();
+            // remove duplicates in file
+            Set<LineTypeInfos> lineTypesSet = lineTypes.stream().collect(Collectors.toSet());
 
-        List<LineTypeEntity> lineTypesEntities = lineTypesSet.stream()
-            .map(LineTypeInfos::toEntity)
-            .collect(Collectors.toList());
-        lineTypesCatalogRepository.saveAll(lineTypesEntities);
+            List<LineTypeEntity> lineTypesEntities = lineTypesSet.stream()
+                .map(LineTypeInfos::toEntity)
+                .collect(Collectors.toList());
+            lineTypesCatalogRepository.saveAll(lineTypesEntities);
+        } catch (IOException e) {
+            throw new UncheckedIOException("reading gzip file error", e);
```

Comment:
other places in the code we just do:
```suggestion
            throw new UncheckedIOException(e);
```

Reply:
ok but other there is a message too but ok as you wish